docs/tutorials/fault_tolerance.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Fault Tolerance via State Snapshots
+nav-id: fault-tolerance
+nav-pos: 6
+nav-title: Fault Tolerance
+nav-parent_id: tutorials
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+* This will be replaced by the TOC
+{:toc}
+
+## State Backends
+
+The keyed state managed by Flink is a sort of sharded, key/value store, and the working copy of each
+item of keyed state is kept somewhere local to the taskmanager responsible for that key. Operator
+state is also local to the machine(s) that need(s) it. Flink periodically takes persistent snapshots
+of all the state and copies these snapshots somewhere more durable, such as a distributed file
+system.
+
+In the event of the failure, Flink can restore the complete state of your application and resume
+processing as though nothing had gone wrong.
+
+This state that Flink manages is stored in a _state backend_. Two implementations of state backends
+are available -- one based on RocksDB, an embedded key/value store that keeps its working state on
+disk, and another heap-based state backend that keeps its working state in memory, on the Java heap.
+This heap-based state backend comes in two flavors: the FsStateBackend that persists its state
+snapshots to a distributed file system, and the MemoryStateBackend that uses the JobManager's heap.
+
+<img src="{{ site.baseurl }}/fig/state-backends.png" alt="State backends" class="center" width="90%" />
+
+When working with state kept in a heap-based state backend, accesses and updates involve reading and
+writing objects on the heap. But for objects kept in the RocksDBStateBackend, accesses and updates
+involve serialization and deserialization, and so are much more expensive. But the amount of state
+you can have with RocksDB is limited only by the size of the local disk. Note also that only the
+RocksDBStateBackend is able to do incremental snapshotting, which is a significant benefit for
+applications with large amounts of slowly changing state.
+
+Both state backends are able to do asynchronous snapshotting, meaning that they can take a snapshot without impeding the ongoing stream processing.
```

Comment:
the reader may be confused here because you had three state backends in the table

Reply:
fixed